relay-server/src/actors/global_config.rs

```diff
@@ -257,23 +261,43 @@ impl Service for GlobalConfigService {
             let mut shutdown_handle = Controller::shutdown_handle();
 
             relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
-                // NOTE(iker): if this first request fails it's possible the default
-                // global config is forwarded. This is not ideal, but we accept it
-                // for now.
-                self.update_global_config();
-            } else {
-                // NOTE(iker): not making a request results in the sleep handler
-                // not being reset, so no new requests are made.
-                relay_log::info!("fetching global configs disabled: no credentials configured");
-            }
+            match self.config.relay_mode() {
+                RelayMode::Managed => {
+                    if self.config.has_credentials() {
+                        relay_log::info!("serving global configs fetched from upstream");
+                        self.request_global_config();
+                    } else {
+                        relay_log::info!(
+                            "serving default global configs due to lacking credentials"
+                        );
+                    }
+                }
+                RelayMode::Static | RelayMode::Proxy | RelayMode::Capture => {
+                    match GlobalConfig::load(self.config.path()) {
+                        Ok(Some(from_file)) => {
+                            relay_log::info!("serving static global config loaded from file");
+                            self.global_config_watch.send(Arc::new(from_file)).ok();
```

Comment:
From the docs on [subscribe](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.subscribe) 
> All messages sent before this call to subscribe are initially marked as seen by the new Receiver.

Can it happen, that e.g.  `EnvelopeProcessorService` will never gets update over the watch, since it will subscribe to the `watch` after this static config is sent? And then we actually never send any other update? 

Or maybe I'm missing something?